### ISSUE TITLE
KMail: missing dependencies

### ISSUE THREAD

**Issue by donottellmetonottellyou** [NixOS/nixpkgs#329423]
### Describe the bug
I installed `kmail` from `kdepackages`. When I launched the email client at first, it crashed because `kmail-account-wizard` wasn't installed. After installing that, it would still crash with an error of `Unable to obtain agent type ''`.

### Steps To Reproduce
1. add kmail to system packages
2. kmail doesn't work out of the box
3. kmail will still crash when adding `kmail-account-wizard`

### Expected behavior
I expect all necessary dependencies of kmail to be installed with the package.

### Screenshots
![image](https://github.com/user-attachments/assets/d7144444-53db-471e-8935-c320128ff112)

### Additional context
- I'm a new-ish user of nix
- KDE Plasma 6 Wayland
- Relevant depends
```nix
      kdepim-addons
      kmail
      kmailtransport
      kmail-account-wizard
      ktexttemplate
```

### Notify maintainers
@K900 @ilya-fedin @LunNova @mjm @NickCao @SuperSandro2000 @ttuegel 

### Metadata
```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.41, NixOS, 24.05 (Uakari), 24.05.3164.63d37ccd2d17`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `"home-manager-24.05.tar.gz, nixgl, nixos-24.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by K900** [comment]
You need to use `programs.kde-pim.enable`.

**Comment by donottellmetonottellyou** [comment]
![image](https://github.com/user-attachments/assets/284cf9c9-5905-41d8-91d4-917bb6be6bcd)

**Comment by K900** [comment]
Currently only available on unstable.

**Comment by K900** [comment]
github.com/NixOS/nixpkgs/pull/299599 can probably be backported though.

**Comment by donottellmetonottellyou** [comment]
It doesn't appear on unstable either?

**Comment by K900** [comment]
It's possible unstable hasn't advanced yet, but it should be there in a few hours.

**Comment by K900** [comment]
https://github.com/NixOS/nixpkgs/pull/330521 for 24.05

**Comment by donottellmetonottellyou** [comment]
This issue has been resolved in the channel `nixos-24.05-small`.

For anyone wanting to install `kmail`, `kontact`, or `merkuro` in Plasma 6, the proper way (once it is released on `nixos-24.05`) is to add the following to your nix config:
```nix
  programs.kde-pim = {
    kmail = true;
    kontact = true;
    merkuro = true;
  };
```
`kde-pim` is enabled by default with `services.desktopManager.plasma6.enable`
